#230 Add full instrumentation for ExecutorService

ZawadaZSE opened the pull request:
Resolves #230 

This merge adds traces propagation to tasks triggered by ExecutorService API.

ZawadaZSE commented:
Ready for review.

pavolloffay commented on instrument-starters/opentracing-spring-cloud-core/src/main/java/io/opentracing/contrib/spring/cloud/async/ExecutorBeanPostProcessor.java:
```diff
@@ -1,11 +1,11 @@
 /**
  * Copyright 2013-2017 the original author or authors. Copyright 2017 The OpenTracing Authors
- *
+ * <p>
```
remove changes in the license header

bono007 commented on instrument-starters/opentracing-spring-cloud-core/src/main/java/io/opentracing/contrib/spring/cloud/async/ExecutorBeanPostProcessor.java:
```diff
@@ -52,49 +80,72 @@ public Object postProcessBeforeInitialization(Object bean, String beanName)
+            (e, t) -> new TracedThreadPoolTaskExecutor(t, e),
+            shouldUseCGLibProxy(threadPoolTaskExecutor, TRACED_THREAD_POOL_TASK_EXECUTOR_METHODS)
+        );
+      } else {
```
@ZawadaZSE how do you feel about adding another block after `ThreadPoolTaskExecutor` for `AsyncTaskExecutor`? It would give proper tracing and cglib proxy for classes such as `ThreadPoolTaskScheduler` . Issue #228 would be fixed by that 

I am also ok adding that in after your PR goes in. 

Thanks

bono007 replied:
@ZawadaZSE what are your thoughts on the above request? Thanks.

pavolloffay commented:
@ZawadaZSE what is status of this PR?

ZawadaZSE commented:
Hey,

sorry for leaving this PR open for so long.
I've removed changes to the license header, and rebased on top of master.

I've checked the implementation of sleuth and it's similar to this one, and I'd like to stick with this one cause in my opinion it's more readable and easier for extension.

I've also fixed issue that might happen if someone would have a bean of final class which extends ThreadPoolTaskExecutor. Previously that would cause BeanNotOfRequiredTypeException as ThreadPoolTaskExecutor is not an interface and JDK proxy won't work, now we won't instrument that bean.

ZawadaZSE commented:
Hey, any progress on this?

pavolloffay commented:
thanks @ZawadaZSE

ZawadaZSE commented:
Thank you :)

Any chance to also release it?

pavolloffay commented:
It has already been released
